src/main/java/de/metas/ui/web/product/process/WEBUI_M_Product_LotNumber_Lock.java

```diff
@@ -50,28 +50,54 @@
  */
 
 /**
- * https://github.com/metasfresh/metasfresh/issues/3693
- * This process will search for HUs containing products with LotNo attributes that fit the pairs in the selected I_M_Product_LotNumber_Lock entries.
- * If such HUs are found, they will all be put in a DD_Order and sent to the Quarantine warehouse.
+ * https://github.com/metasfresh/metasfresh/issues/3693 This process will search
+ * for HUs containing products with LotNo attributes that fit the pairs in the
+ * selected I_M_Product_LotNumber_Lock entries. If such HUs are found, they will
+ * all be put in a DD_Order and sent to the Quarantine warehouse.
  * 
  * @author metas-dev <dev@metasfresh.com>
  *
  */
-public class WEBUI_M_Product_LotNumber_Lock extends ViewBasedProcessTemplate implements IProcessPrecondition
+public class WEBUI_M_Product_LotNumber_Lock extends ViewBasedProcessTemplate
+		implements
+		IProcessPrecondition
 {
-	private final IInvoiceCandBL invoiceCandBL = Services.get(IInvoiceCandBL.class);
+	private final IInvoiceCandBL invoiceCandBL = Services
+			.get(IInvoiceCandBL.class);
 	private final IHUInOutDAO huInOutDAO = Services.get(IHUInOutDAO.class);
 
+	private List<HUToDistribute> husToDistribute = new ArrayList<>();
+
 	@Override
 	protected String doIt() throws Exception
 	{
 		getView().streamByIds(getSelectedRowIds())
-				.map(row -> row.getId().toInt())
-				.distinct()
-				.forEach(id -> quarantineHUsForLotNo(id));
+				.map(row -> row.getId().toInt()).distinct()
+				.forEach(id -> createQuarantineHUs(id));
+
+		createQuarantineDDOrders();
+		setInvoiceCandsInDispute();
+
 		return MSG_OK;
 	}
 
+	private void createQuarantineDDOrders()
+	{
+		Services.get(IHUDDOrderBL.class)
+				.createQuarantineDDOrderForHUs(husToDistribute);
+
+	}
+
+	private void setInvoiceCandsInDispute()
+	{
+		husToDistribute.stream().map(HUToDistribute::getHu).collect(ImmutableList.toImmutableList())
+				.stream()
+				.flatMap(hu -> huInOutDAO.retrieveInOutLinesForHU(hu).stream())
+				.forEach(line -> {
+					invoiceCandBL.markInvoiceCandInDisputeForReceiptLine(line);
+				});
```

Comment:
you can write that one in one line. You can even use the `::` operator

Reply:
Done